client/src/components/LoginSession/tests/LoginSessionContainer-test.js

```diff
@@ -28,40 +39,153 @@ window.ss.config = {
     SecurityID: 1234567890
 };
 
-describe('LoginSessionContainer', () => {
-    let wrapper;
-    let props;
+const sessionData = {
+    IPAddress: '127.0.0.1',
+    IsCurrent: false,
+    UserAgent: 'Chrome on Mac OS X 10.15.7',
+    Created: '2021-01-20 00:33:41',
+    LastAccessed: '2021-03-11 03:47:22'
+};
+
+const props = {
+    // used in LoginSessionContainer
+    ID: 1,
+    LogOutEndpoint: '/test',
+    displayToastSuccess: jest.fn(() => 1),
+    displayToastFailure: jest.fn(() => 1),
+    // passed on to LoginSession
+    ...sessionData
+};
+
+MockDate.set('2021-03-12 03:47:22');
 
+describe('LoginSessionContainer', () => {
     beforeEach(() => {
-        MockDate.set('2021-03-12 03:47:22');
-        props = {
-            // used in LoginSessionContainer
-            ID: 1,
-            LogOutEndpoint: '/test',
-            displayToastSuccess: jest.fn(() => 1),
-            displayToastFailure: jest.fn(() => 1),
-            // passed on to LoginSession
-            IPAddress: '127.0.0.1',
-            IsCurrent: false,
-            UserAgent: 'Chrome on Mac OS X 10.15.7',
-            Created: '2021-01-20 00:33:41',
-            LastAccessed: '2021-03-11 03:47:22',
-        };
+        props.displayToastSuccess.mockClear();
+        props.displayToastFailure.mockClear();
     });
 
     it('should call displayToastSuccess on success', async () => {
         setEndpointFetcherSuccess(true);
-        wrapper = shallow(<LoginSessionContainer {...props} />);
-        await wrapper.instance().logout();
-        expect(props.displayToastSuccess).toBeCalledTimes(1);
-        expect(props.displayToastFailure).toBeCalledTimes(0);
+        let wrapper = shallow(<LoginSessionContainer {...props} />);
+        let loginSession = wrapper.find(LoginSession).first();
+
+        expect(loginSession.props()).toMatchObject({
+            ...sessionData,
+            complete: false,
+            failed: false,
+            submitting: false
+        });
+
+        const logoutRequest = loginSession.props().logout();
```

Comment:
We extract the `logout()` handler from `<LoginSession />` and manually call it to trigger a state change in `<LoginSessionContainer />`.